controller/src/main/java/com/flipkart/varadhi/controller/SuppressionManager.java

```diff
@@ -0,0 +1,108 @@
+package com.flipkart.varadhi.controller;
+
+import com.flipkart.varadhi.qos.entity.ClientHistory;
+import com.flipkart.varadhi.qos.entity.ClientLoadInfo;
+import com.flipkart.varadhi.qos.entity.SuppressionData;
+import com.flipkart.varadhi.qos.entity.SuppressionFactor;
+import com.flipkart.varadhi.qos.entity.TopicLoadInfo;
+import com.flipkart.varadhi.utils.FutureUtil;
+import com.google.common.base.Ticker;
+
+import java.util.ArrayList;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.mutable.MutableDouble;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+public class SuppressionManager implements SuppressionService {
```

Comment:
This class needs to be async. There can be wrappers on this class that may convert this to async, but this class should be made sync (at least in current impl)

Reply:
Made everything sync